tests/ImageSharp.Benchmarks/Codecs/DecodeTiff.cs

```diff
@@ -1,38 +1,94 @@
 // Copyright (c) Six Labors.
 // Licensed under the Apache License, Version 2.0.
 
+//// #define BIG_TESTS
+
 using System.IO;
+
 using BenchmarkDotNet.Attributes;
+
+using SixLabors.ImageSharp.Formats.Experimental.Tiff;
 using SixLabors.ImageSharp.PixelFormats;
 using SixLabors.ImageSharp.Tests;
+
 using SDImage = System.Drawing.Image;
 using SDSize = System.Drawing.Size;
 
 namespace SixLabors.ImageSharp.Benchmarks.Codecs
 {
-    [Config(typeof(Config.ShortClr))]
+    [Config(typeof(Clr))]
     public class DecodeTiff : BenchmarkBase
     {
-        private byte[] tiffBytes;
+        public class Clr : Config.ShortClr
+        {
+            public Clr() => this.SummaryStyle = BenchmarkDotNet.Reports.SummaryStyle.Default.WithMaxParameterColumnWidth(60);
+        }
+
+        private string prevImage = null;
+
+        private byte[] data;
+
+        private Configuration configuration;
+
+#if BIG_TESTS
+        private static readonly int BufferSize = 1024 * 68;
+
+        private string TestImageFullPath => Path.Combine(TestEnvironment.InputImagesDirectoryFullPath, Path.Combine(TestImages.Tiff.Benchmark_Path, this.TestImage));
+
+        [Params(
+            TestImages.Tiff.Benchmark_BwFax3,
+            //// TestImages.Tiff.Benchmark_RgbFax4,
+            TestImages.Tiff.Benchmark_BwRle,
+            TestImages.Tiff.Benchmark_GrayscaleUncompressed,
+            TestImages.Tiff.Benchmark_PaletteUncompressed,
+            TestImages.Tiff.Benchmark_RgbDeflate,
+            TestImages.Tiff.Benchmark_RgbLzw,
+            TestImages.Tiff.Benchmark_RgbPackbits,
+            TestImages.Tiff.Benchmark_RgbUncompressed)]
+        public string TestImage { get; set; }
+
+#else
+        private static readonly int BufferSize = Configuration.Default.StreamProcessingBufferSize;
 
         private string TestImageFullPath => Path.Combine(TestEnvironment.InputImagesDirectoryFullPath, this.TestImage);
 
-        [Params(TestImages.Tiff.RgbPackbits)]
+        [Params(
+            TestImages.Tiff.CcittFax3AllTermCodes,
+            TestImages.Tiff.HuffmanRleAllMakeupCodes,
+            TestImages.Tiff.GrayscaleUncompressed,
+            TestImages.Tiff.PaletteUncompressed,
+            TestImages.Tiff.RgbDeflate,
+            TestImages.Tiff.RgbLzwPredictor,
+            TestImages.Tiff.RgbPackbits,
+            TestImages.Tiff.RgbUncompressed)]
         public string TestImage { get; set; }
+#endif
 
         [GlobalSetup]
+        public void Config()
+        {
+            if (this.configuration == null)
+            {
+                this.configuration = new Configuration();
+                this.configuration.AddTiff();
+                this.configuration.StreamProcessingBufferSize = BufferSize;
+            }
+        }
+
+        [IterationSetup]
```

Comment:
I think the image bytes need to be loaded during `GlobalSetup` not in `IterationSetup`. Maybe do it the same way as for example `DecodeFilteredPng.cs`

Reply:
Yep, definitely global.

Reply:
is actually only loaded when needed: https://github.com/SixLabors/ImageSharp/pull/1496/files/d83f32704a669d3472685fdbb381c401240974f1#diff-3243ee1e2090c5162ac369f99493f7b1cccd702d463fdbc40880f44ced0474d0R81

Several files are used for testing, so that not all files are loaded into memory at once, only the current test file is loading.